evm/vm/base.py

```diff
@@ -712,7 +713,8 @@ def validate_block(self, block):
             )
 
     @classmethod
-    def validate_header(cls, header: BlockHeader, parent_header: BlockHeader) -> None:
+    def validate_header(
+            cls, header: BlockHeader, parent_header: BlockHeader, check_seal: bool) -> None:
```

Comment:
I think it's okay to default `check_seal` to `True`, which should remove some of the boilerplate changes.